components/prombench/manifests/benchmark/prometheus-pr.yaml

```diff
@@ -0,0 +1,253 @@
+        target_label: job
+      - action: keep
+        source_labels: [__meta_kubernetes_service_name]
+        regex: ".*-prometheus-discovery"
+      - action: keep
+        source_labels: [__meta_kubernetes_endpoint_port_name]
+        regex: "http-metrics.*|https-metrics.*"
+      - action: replace
+        source_labels: [__meta_kubernetes_endpoint_port_name]
+        regex: "https-metrics.*"
+        target_label: __scheme__
+        replacement: https
+      - action: replace
+        source_labels: [__meta_kubernetes_endpoint_port_name]
+        regex: "https-metrics.*"
+        target_label: __scheme__
+        replacement: https
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: prometheus-test-{{ .PROMETHEUS_1_NAME }}
+  namespace: prombench-{{ .PR_NUMBER }}
+  labels:
+    app: prometheus
+    prometheus: test-{{ .PROMETHEUS_1_NAME }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: prometheus
+      prometheus: test-{{ .PROMETHEUS_1_NAME }}
+  template:
+    metadata:
+      namespace: prombench-{{ .PR_NUMBER }}
+      labels:
+        app: prometheus
+        prometheus: test-{{ .PROMETHEUS_1_NAME }}
+    spec:
+      serviceAccountName: prometheus
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - topologyKey: kubernetes.io/hostname
+            labelSelector:
+              matchExpressions:
+              - key: app
+                operator: In
+                values:
+                - prometheus
+      securityContext:
+        runAsUser: 0
+      containers:
+      - name: prometheus
+        image: {{ .PROMETHEUS_1_IMAGE }}
+        imagePullPolicy: Always
+        args: [
+          "--config.file=/etc/prometheus/config/prometheus.yaml",
+          "--storage.tsdb.path=/data"
+        ]
+        volumeMounts:
+        - name: config-volume
+          mountPath: /etc/prometheus/config
+        - name: instance-ssd
+          mountPath: /data
+      volumes:
+      - name: config-volume
+        configMap:
+          name: prometheus-test
+      - name: instance-ssd
+        hostPath:
+          # /mnt is where GKE keeps it's SSD
+          # don't change this if you want Prometheus to take advantage of these local SSDs
+          path: /mnt/disks/ssd0
+      terminationGracePeriodSeconds: 300
+      nodeSelector:
+        cloud.google.com/gke-nodepool: prometheus-{{ .PR_NUMBER }}
+        isolation: prometheus
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: prometheus-test-{{ .PROMETHEUS_1_NAME }}
+  namespace: prombench-{{ .PR_NUMBER }}
+  labels:
+    app: prometheus
+    prometheus: test-{{ .PROMETHEUS_1_NAME }}
+spec:
+  ports:
+  #should be 1 port
+  - name: prometheus
+    port: 9090
+  selector:
+    app: prometheus
+    prometheus: test-{{ .PROMETHEUS_1_NAME }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: prometheus-test-{{ .PROMETHEUS_2_NAME }}
```

Comment:
Lets try to keep the env variables to a minimum to make it easyer to understand what is happening.
I think it is fine to hard core PROMETHEUS_1_IMAGE and PROMETHEUS_2_IMAGE to make it easyer to understand .

Reply:
The values of these args are not constant are are dependent on /benchmark pr|release

Reply:
these are only used in w places